### PR TITLE
internal: add Field primitive

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

Adding an internal Field primitive for custom fields like a StorageManagerField

```jsx
    <Field
      label="Images"
      isReadOnly={false}
      isRequired={false}
      errorMessage={'error'}
      hasError={false}
    >
      <StorageManager
        acceptedFileTypes={['image/*']}
        accessLevel="private"
        maxFileCount={3}
        isResumable
        showThumbnails={true}
      />
    </Field>
```

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by calebpollman — APPROVED** [review]
🚢

**Review by calebpollman — DISMISSED** [review]

File: packages/react/src/primitives/Field/Field.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      {children}
+      {errorMessage ? (
+        <FieldErrorMessage hasError={hasError} errorMessage={errorMessage} />
+      ) : null}
+    </Flex>
+  );
+};
+
+/**
+ * [📖 Docs](https://ui.docs.amplify.aws/react/components/field)
```
Comment (reesscot) [inline packages/react/src/primitives/Field/Field.tsx:67]: This page doesn't actually exist right?

File: packages/ui/src/theme/css/component/textField.scss [hunk 2]
```diff
@@ -1,5 +1,4 @@
 .amplify-textfield {
-  flex-direction: column;
```
Comment (reesscot) [inline packages/ui/src/theme/css/component/textField.scss:2]: This moved to Field styles, is that what I'm gathering?

Comment (dbanksdesign) [reply packages/ui/src/theme/css/component/textField.scss:2]: that is correct

**Review by reesscot — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**